### ISSUE TITLE
Updating existing empty map results in single-line yaml syntax

### ISSUE THREAD

**Issue by marcostvz** [mikefarah/yq#1176]
**Describe the bug**
As described in https://mikefarah.gitbook.io/yq/operators/assign-update#update-empty-object-and-array, updating a file with an existing empty map results in a one-liner yaml. In our use case, this makes helm template to break.

Version of yq: 4.24.2
Operating system: tested on mac and linux
Installed via: homebrew for mac, binary release for linux

**Input Yaml**
fields.yaml:
```
global: {}
```

**Command**
```
yq eval '.global.test = "val"' fields.yaml
```

**Actual behavior**
```
global: {test: val}
```

**Expected behavior**
```
global:
  test: val
```

**Additional context**
If we work with an empty file, or a non-existing key, the format is correct:
```
$ yq eval '.global.test = "val"' empty-fields.yaml
global:
  test: val

$ yq eval '.other.test = "val"' fields.yaml
global: {}
other:
  test: val
```

Also, as a workaround, we can parse it to json and back to yaml and the format is correct:
```
$ yq eval '.global.test = "val"' fields.yaml -o json | yq -P '.' -
global:
  test: val
```

**Comment by mikefarah** [comment]
Yeah this is a side effect of `yq` trying to maintain the styling used in the document. I'm surprised helm doesn't like it as it's perfectly valid yaml.

I'll look into making it a bit smarter for scenarios like this, till then you can:

```
yq -P '.global.test = "val"' fields.yaml
```

**Comment by mikefarah** [comment]
Fixed in 4.24.5

**Comment by krokofant** [comment]
I want to have the empty map converted to a single line definition when updated. Seems there is no way to achieve this now.

I have branches with commits changing something like:
```yaml
foo1:
  bar: true
```
and 
```yaml
foo2:
  bar: true
```

Now merging these automatically would result in (depending on which way)
```yaml
foo1:
foo2:
  bar: true
```

If I could maintain the yaml in a single line, then I could work around this issue. The merge would be:
```
foo1: { bar: true }
foo2: { bar: true }
```

I expected combining these commands would preserve the inline format:
```shell
yq e -i ".foo1 = {}" myvalues.yaml
yq e -i ".foo1.bar = true" myvalues.yaml
```
